x/derivatives/types/pool.go

```diff
@@ -62,3 +62,10 @@ func IsValidDepositForPool(deposit sdk.Coin, acceptableAssets []PoolAssetConf) b
 	}
 	return false
 }
+
+// CalcTargetAmountInPool is used to calculate the target amount of the asset in the pool.
+// The target amount is calculated by the formula:
+// targetAmount = poolMarketCap * weight / price
+func CalcTargetAmountInPool(weight, price, poolMarketCap sdk.Dec) sdk.Int {
+	return poolMarketCap.Mul(weight).Quo(price).TruncateInt()
```

Comment:
I think price zero case should be considered

Reply:
Yes, I def agree **in general.**
that's why I created this https://github.com/UnUniFi/chain/issues/446.

But, if we use sdk.Int or sdk.Dec for the calculation of the division, it will cause panic if the dividing num is zero, right?
So, I think this issue will be better to be addressed in a more structural way for all the other part at once. 
What do you think?